python/test/unit/language/test_matmul.py

```diff
@@ -845,24 +872,61 @@ def mxfp8_mxfp4_matmul(  #
 @pytest.mark.parametrize("NUM_STAGES", [1, 3])
 @pytest.mark.parametrize("B_TRANS", [True, False])
 @pytest.mark.parametrize("CONST_SCALE", [True, False])
-@pytest.mark.skipif(torch.cuda.get_device_capability()[0] < 10, reason="Requires compute capability >= 10")
-def test_mxfp8_mxfp4_matmul(M, N, K, BLOCK_M, BLOCK_N, BLOCK_K, NUM_STAGES, B_TRANS, CONST_SCALE, device):
-    if BLOCK_N == 256 and BLOCK_K == 256:
+@pytest.mark.parametrize("A_DATA_TYPE", ["float8e5", "float8e4nv", "float4"])
+@pytest.mark.parametrize("B_DATA_TYPE", ["float8e5", "float8e4nv", "float4"])
+@pytest.mark.parametrize("WITH_A_SCALE", [True, False])
+@pytest.mark.parametrize("WITH_B_SCALE", [True, False])
+@pytest.mark.parametrize("nonKDim", ([0, 16, 32] if is_hip_cdna() else []))
+def test_mxfp8_mxfp4_matmul(M, N, K, BLOCK_M, BLOCK_N, BLOCK_K, NUM_STAGES, B_TRANS, CONST_SCALE, A_DATA_TYPE,
+                            B_DATA_TYPE, WITH_A_SCALE, WITH_B_SCALE, nonKDim, device):
+    if is_cuda():
+        if torch.cuda.get_device_capability()[0] < 10:
+            pytest.skip("Requires compute capability >= 10")
+        if not (WITH_A_SCALE and WITH_B_SCALE):
+            pytest.skip("None scale has not been tested on NV backend")
+        if not (A_DATA_TYPE == "float8e5" and B_DATA_TYPE == "float4"):
+            pytest.skip(f"(A: {A_DATA_TYPE}, B: {B_DATA_TYPE}) has not been tested on NV backend")
+    elif is_hip():
+        if not is_hip_mi350():
+            pytest.skip("Scaled mxfp4 & mxfp8 matmul is only natively supported on MI350")
+        if CONST_SCALE:
+            pytest.skip("Constant scale is not supported in AMD backend for now")
+        if (nonKDim == 16 and BLOCK_K < 128) or (nonKDim == 32 and BLOCK_K < 64):
+            pytest.skip(f"MI350 does not support {BLOCK_K=} for scaled mfma {nonKDim=} variants")
+        if (A_DATA_TYPE == 'float4' and not WITH_A_SCALE) or (B_DATA_TYPE == 'float4' and not WITH_B_SCALE):
+            pytest.skip("Float4 without scale is tested in other testcases")
```

Comment:
Nit: be explicit with the testcase name?

Also fp8 w/o scale is also tested in `test_core.py` `test_scaled_dot`? Would those be sufficient already?

Reply:
Does "scale is None" refer to the case that all scale values are 1 (127 in e8m0 value)?

Reply:
Yes. If it's None, we will create such a constant scale.

Reply:
Okay I see. This is fine then.